lib/puppet/provider/service/smf.rb

```diff
@@ -121,33 +143,26 @@ def restart
     self.wait('online')
   end
 
-  # Determine the current and next states of a service.
-  def service_states
-    svcs("-H", "-o", "state,nstate", @resource[:name]).chomp.split
-  end
-
   def status
-    if @resource[:status]
-      super
-      return
-    end
+    # TODO (PUP-8957): This originally returned nil after calling super, which is inconsistent
+    # with how things are done.
+    return super if @resource[:status]
```

Comment:
Is this fine? From what I could tell, super's code here https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/service/base.rb#L68-L85 returned the status so it seemed odd that the original code didn't.